kube-core/src/labels.rs

```diff
@@ -0,0 +1,490 @@
+#![allow(missing_docs)]
+use core::fmt;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::{LabelSelector, LabelSelectorRequirement};
+use serde::{Deserialize, Serialize};
+use std::{
+    cmp::PartialEq,
+    collections::{BTreeMap, BTreeSet},
+    fmt::Display,
+    iter::FromIterator,
+    option::IntoIter,
+};
+
+use crate::ResourceExt;
+
+// local type aliases
+type Map = BTreeMap<String, String>;
+type Expressions = Vec<Expression>;
+
+/// Extensions to [`ResourceExt`](crate::ResourceExt)
+/// Helper methods for resource selection based on provided Selector
+pub trait SelectorExt {
```

Comment:
i wonder if it's worth considering a method here that allows you to create selectors from an object and a single label so that you could do some quick stuff like:

```
let o: Pod = todo!();
 // create a selector from `Expression::equal` using given key name and objects label value
let selector = o.selector_from_label("environment")?;
ListParams::default().labels_from(&selector)
```

Reply:
I feel like this needs separate parsing capabilities which converge to a specific type, `Expression` in this case. I added some docs with examples for the `Expression` struct. These can be used standalone, as it implements the `Matcher` trait, or convert to `Selector`. Should it be enough? I think more functionality can be added later, and extend the trait if needed.

```rust
let matches = Pod::default().selector_matches(Expression::Exists("foo".into()));
// or
let wp = WatchParams::default().labels_from(&Expression::Exists("foo".into()).into())
```

Reply:
Sounds good. let's keep it simple for now.